### PR TITLE
added serializers and tests for the website collection related objects

### LINKED ISSUE #334: add serializers for updating WebsiteCollection and WebsiteCollectionItem models
We need serializers!

these are just a little more complicated than normal because of the `position` property on the `WebsiteCollectionItems`. I'm going to base these on the code we already wrote for handling very similar things in Open Discussions.

### PR DESCRIPTION
**Pull request by aliceriot**
#### Pre-Flight checklist

- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?

closes #334

#### What's this PR do?

This adds serializers for the `WebsiteCollection` and `WebsiteCollectionItem` models.

#### How should this be manually tested?

hm, we'll I suppose it's not so easy to test this manually - I think you could do so in the django shell if you wanted. I mainly just wanted to keep my PR from getting too big, so I pulled this out into a separate PR. Will be doing another shortly to add the views to actually get APIs going with these.

But definitely read through the code carefully and make sure it makes sense!

### CODE REVIEW

File: websites/serializers.py [hunk 1]
```diff
@@ -261,3 +268,72 @@ class Meta:
...
+
+
+class WebsiteCollectionSerializer(serializers.ModelSerializer):
+    """Serializer for WebsiteCollections"""
+
+    item_count = serializers.SerializerMethodField(read_only=True)
+
+    def get_item_count(self, instance):
+        """get the number of items in this collection"""
+        return WebsiteCollectionItem.objects.filter(website_collection=instance).count()
```
Comment (noisecapella) [inline websites/serializers.py:280]: This could be an n+1 query but it may not matter. Are there likely to be a lot of `WebsiteCollection`s?

Comment (aliceriot) [reply websites/serializers.py:280]: I'm not sure, I think at least a few

I wasn't sure actually if we even needed this field, or if it maybe would be better to set it as a value on the model whenever we add or remove a `WebsiteCollectionItem`. I was also wondering if maybe this is something where I would need to do something with the `get_queryset` method on the associated view in order to eliminate possible issues - what do you think?

Comment (aliceriot) [reply websites/serializers.py:280]: I think I'll just take out the 'num_items' thing for now to keep it simple

**Comment by aliceriot** [comment]
@noisecapella I updated it to remove that thing for now

**Review by noisecapella — APPROVED** [review]

**Pull request merged**